### PR TITLE
[SPARK-7639] [PySpark] [MLlib] Python API for KernelDensity

### PR DESCRIPTION
**Pull request by MechCoder**
Python API for KernelDensity

### CODE REVIEW

**Comment by MechCoder** [comment]
Will add tests as soon as https://github.com/apache/spark/pull/6383 is resolved (or) merged.

cc @mengxr

File: python/pyspark/mllib/stat/KernelDensity.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+class KernelDensity(object):
+    """
+    .. note:: Experimental
+
+    Estimate probabiltiy density at required points given a sample from the
+    population.
+    """
+    def __init__(self):
+        self._bandwidth = 1.0
+        self._sample = 0
```
Comment (mengxr) [inline python/pyspark/mllib/stat/KernelDensity.py]: The default should be `None`.

File: python/pyspark/mllib/stat/KernelDensity.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+    def setBandwidth(self, bandwidth):
+        self._bandwidth = bandwidth
+
+    def setSample(self, sample):
+        self._sample = sample
```
Comment (mengxr) [inline python/pyspark/mllib/stat/KernelDensity.py]: Raise an error if sample is not an RDD.

File: mllib/src/main/scala/org/apache/spark/mllib/stat/KernelDensity.scala [hunk 3]
```diff
@@ -17,6 +17,7 @@
 
 package org.apache.spark.mllib.stat
 
+import java.util.ArrayList
```
Comment (mengxr) [inline mllib/src/main/scala/org/apache/spark/mllib/stat/KernelDensity.scala]: Remove unnecessary imports.

File: python/pyspark/mllib/stat/KernelDensity.py [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+from pyspark.mllib.common import callMLlibFunc
+
+class KernelDensity(object):
+    """
+    .. note:: Experimental
+
+    Estimate probabiltiy density at required points given a sample from the
+    population.
+    """
```
Comment (mengxr) [inline python/pyspark/mllib/stat/KernelDensity.py]: Add a doctest.

**Comment by MechCoder** [comment]
@mengxr done!

**Comment by mengxr** [comment]
@MechCoder Python doctests are not automatically discovered. We have to include them manually. See: https://github.com/apache/spark/blob/master/python/run-tests#L93. Please include `KernelDensity` there and let Jenkins test against different Python versions.

**Comment by MechCoder** [comment]
done !

**Comment by mengxr** [comment]
LGTM (I'm going to merge this after 1.4.)

**Comment by mengxr** [comment]
Merged into master. Thanks!

**Pull request closed without merging**